Zend/tests/closure_const_expr/attributes.phpt

```diff
@@ -0,0 +1,57 @@
+--TEST--
+Allow defining closures in attributes
+--EXTENSIONS--
+reflection
+--FILE--
+<?php
+
+#[Attribute(Attribute::TARGET_CLASS | Attribute::IS_REPEATABLE)]
+class Attr {
+    public function __construct(public Closure $value) {
+      $value('foo');
+    }
+}
+
+#[Attr(static function () { })]
```

Comment:
One thing that wasn't clear to me from the RFC or the code is what the scope of attributes of classes are (it is in fact the class they are declared on). This should be tested, and possibly clarified in the RFC. This means, calling `private` static methods of the class from the closure should be possible.

Reply:
> This should be tested,

It is in `Zend/tests/closure_const_expr/attributes_scope_001.phpt` (with a property, not a method).

> and possibly clarified in the RFC.

It's mentioned in the RFC “Likewise are Closures in attribute parameters allowed to access private members of the surrounding class.” https://wiki.php.net/rfc/closures_in_const_expr#scoping

Reply:
> It is in Zend/tests/closure_const_expr/attributes_scope_001.phpt (with a property, not a method).

Thank you, I missed that.

> It's mentioned in the RFC “Likewise are Closures in attribute parameters allowed to access private members of the surrounding class.”

It wasn't obvious whether "surrounding" includes the class for class attributes. But I think the behavior you picked makes the most sense.